src/composables/useSharing.ts

```diff
@@ -35,38 +41,65 @@ export function useSharing() {
     });
   }
 
-  function shareVote(space, proposal, choices: string) {
-    const handle = space.twitter ? `@${space.twitter}` : space.name;
-    const isSingleChoice =
-      proposal.type === 'single-choice' || proposal.type === 'basic';
-    const isPrivate = proposal.privacy === 'shutter';
-    const text =
-      isSingleChoice && !isPrivate
-        ? `I just voted "${choices}" on`
-        : `I just voted on`;
+  function shareVote(
+    shareTo: 'twitter' | 'lenster',
+    payload: { space: ExtendedSpace; proposal: Proposal; choices: string }
+  ) {
+    const postText = getSharingText(shareTo, payload);
+
     if (isSupported.value)
-      share({
+      return share({
         title: '',
-        text: `${text} "${proposal.title}" ${handle} @SnapshotLabs`,
-        url: proposalUrl(space.id, proposal)
+        text: postText,
+        url: proposalUrl(payload.space.id, payload.proposal)
       });
-    else if (window) {
-      shareTwitter(
-        `${encodeURIComponent(text)}%20"${encodeURIComponent(
-          proposal.title
-        )}"%20${encodedProposalUrl(
-          space.id,
-          proposal
-        )}%20${handle}%20@SnapshotLabs`
-      );
-    }
+    if (window && shareTo === 'twitter') return shareTwitter(postText);
+    if (window && shareTo === 'lenster') return shareLenster(postText);
+  }
+
+  function getSharingText(shareTo: 'twitter' | 'lenster', payload): string {
+    const isSingleChoice =
+      payload.proposal.type === 'single-choice' ||
+      payload.proposal.type === 'basic';
+    const isPrivate = payload.proposal.privacy === 'shutter';
+    const votedText =
+      payload.choices && isSingleChoice && !isPrivate
+        ? `I just voted "${payload.choices}" on`
+        : `I just voted on`;
+
+    const spaceHandle = payload.space.twitter
+      ? `@${payload.space.twitter}`
+      : payload.space.name;
+
+    if (isSupported.value)
+      return `${votedText} "${payload.proposal.title}" ${spaceHandle} @SnapshotLabs`;
+    if (shareTo === 'twitter')
+      return `${encodeURIComponent(votedText)}%20"${encodeURIComponent(
+        payload.proposal.title
+      )}"%20${encodedProposalUrl(
+        payload.space.id,
+        payload.proposal
+      )}%20${spaceHandle}%20@SnapshotLabs`;
+    if (shareTo === 'lenster')
+      return `${encodeURIComponent(votedText)}%20"${encodeURIComponent(
+        payload.proposal.title
+      )}"%20${encodedProposalUrl(
+        payload.space.id,
+        payload.proposal
+      )}&hashtags=snapshotlabs`;
```

Comment:
You can also have a `preview` flag if needed, so it will be not editable 🙇🏼 

<img width="878" alt="Screenshot 2022-10-28 at 3 51 22 PM" src="https://user-images.githubusercontent.com/69431456/198564957-cd54ece5-b942-4050-b87d-8e496a3bdc09.png">

Reply:
I would rather leave it editable @bigint because I couldn't find a way to exit preview mode.

Reply:
I see now that there is a preview icon, but this wasn't obvious to me when I tested it. Maybe it would help to add a background to the icon when it's active.